compiler/src/dotty/tools/dotc/parsing/Parsers.scala

```diff
@@ -1714,7 +1722,7 @@ object Parsers {
      *                  |  override
      *  LocalModifier  ::= abstract | final | sealed | implicit | lazy
      */
-    def modifiers(allowed: BitSet = modifierTokens, start: Modifiers = Modifiers()): Modifiers = {
+    def modifiers(allowed: BitSet = modifierTokens, start: Modifiers = Modifiers(), parsedSeparately: BitSet = BitSet.empty): Modifiers = {
```

Comment:
I am not convinced by this change of API. If the caller wants to parse some modifier separately, then he can exclude them from the allowed modifiers.

Reply:
To get the nicer error I need to distinguish "not allowed" (which gives an error) from "parsed separately" (which doesn't). I had to use `parsedSeparately` only once.

Reply:
If the API is too hard for the error, I can revert/separate these refactorings.

Reply:
I think a better API would something along the line of:
```scala
/** ...
 *  @param errorOn emit a syntax error when a parsed modifier is in errorOn
 */
def modifiers(allowed: BitSet, start: Modifiers, errorOn: BitSet) = {
  ...
  if (allowed.contains(in.token) { ...; loop() }
  else if (errorOn.contains(in.token) { syntaxError; loop() }
  else ...
}
```
You can keep parsing modifiers after encountering an invalid modifier.

But since it is used only once, I think we should no complicate this API.